Fix PHP errors and warnings

yardenac opened the pull request:
Not sure what PHP versions you've been testing on, but it won't run on PHP7.3 without these

yardenac committed:
fix open_basedir warnings

if .git/HEAD isn't in open_basedir it'd throw ugly warnings
suppress errors while checking if file is readable

logmanoriginal commented on lib/Configuration.php:
```diff
@@ -107,7 +107,9 @@ public static function getVersion() {
-		if(file_exists($headFile)) {
+		$is_readable = @is_readable($headFile);
+
+		if($is_readable && file_exists($headFile)) {
```
Checking for `is_readable` makes sense, good point :+1: 

However, it doesn't make sense to use `is_readable` and `file_exists` on the same subject, because `is_readable` already checks if the file exists:

> is_readable — Tells whether a file exists and is readable
>
> -- http://php.net/manual/en/function.is-readable.php

logmanoriginal commented on lib/Configuration.php:
```diff
@@ -107,7 +107,9 @@ public static function getVersion() {
 		$headFile = '.git/HEAD';
 
-		if(file_exists($headFile)) {
+		$is_readable = @is_readable($headFile);
```
The variable `$is_readable` is not necessary (see comment below).
Please add a short comment to why `@` is needed here.

i.e. https://github.com/RSS-Bridge/rss-bridge/blob/1cfe939927ca2f62c063ba55b656f7211a5d9c6d/lib/contents.php#L196

logmanoriginal reviewed (changes requested):
Thanks for this PR!

Find below a few comments on the code regarding handling the .git files. Please split the commit for `simplehtmldom`, because we don't accept PRs for vendor projects (more below).

>Not sure what PHP versions you've been testing on, but it won't run on PHP7.3 without these

Currently 5.6, 7.0, hhvm and nightly (which is 7.3 RC4 [right now](https://wiki.php.net/todo/php73)). See https://travis-ci.org/RSS-Bridge/rss-bridge
The tests don't fail for `simplehtmldom` because it is responsibility of the [vendor project](https://sourceforge.net/projects/simplehtmldom/).

>newer PHP interprets the hyphen as a range and says 'Compilation failed: invalid range in character class'
so we need to escape it
>
>--- https://github.com/RSS-Bridge/rss-bridge/pull/887/commits/e1e43fcfaea44c3d196f7a635c336bec9a3f7e13

Thanks for reporting this issue, your changes absolutely make sense to me. :+1: 

That being said, sources in the `vendor` directory should be fixed in the vendor specific repository first and pulled into RSS-Bridge second. We don't want to diverge from their master. Please open a ticket on SourceForge: https://sourceforge.net/projects/simplehtmldom/

Unfortunately the `PHP Simple HTML DOM Parser` project has been inactive since 2012, so we need to figure out a way to make this work...

yardenac committed:
less redundant, with comment

as requested by the maintainers

yardenac commented:
Ok, I've fixed the open_basedir commit as requested and removed the simplehtmldom commit.

As for simplehtmldom, you're right, the prospects don't look great. They already have an old ticket asking for PHP7 compatibility. Meanwhile I've created a fork [here](https://github.com/yardenac/simplehtmldom). Feel free to use that as "upstream" instead of sourceforge. Yay for open source licenses.

logmanoriginal commented:
Merged. Thanks :+1: 

Regarding simplehtmldom I would like some additional input from the community. I'll open an issue based on your input. Thanks again for raising this issue!
